bgp/routes.go

```diff
@@ -0,0 +1,281 @@
+	// <available | advertised>
+	if args[0] != availableKW && args[0] != advertisedKW {
+		return fmt.Errorf("invalid argument: `%s`, expected `%s` / `%s`", args[0], availableKW, advertisedKW)
+	}
+
+	// <afi> <safi>
+	if len(args) < 2 {
+		return fmt.Errorf("missing AFI value (e.g. `%s`)", ipv4AFI)
+	}
+	if types.ParseAfi(args[1]) == types.AfiUnknown {
+		return fmt.Errorf("unknown AFI `%s`", args[1])
+	}
+	if len(args) < 3 {
+		return fmt.Errorf("missing SAFI value (e.g. `%s`)", unicastSAFI)
+	}
+	if types.ParseSafi(args[2]) == types.SafiUnknown {
+		return fmt.Errorf("unknown SAFI `%s`", args[2])
+	}
+	if len(args) > 3 && (args[3] != vRouterKW && args[3] != peerKW && args[3] != neighborKW) {
+		return fmt.Errorf("invalid argument: `%s`, expected `%s` / `%s`", args[3], vRouterKW, peerKW)
+	}
+	checkArgs := args[3:] // re-slice processed arguments
+
+	// [vrouter <asn>]
+	if len(checkArgs) > 0 && checkArgs[0] == vRouterKW {
+		if len(checkArgs) < 2 {
+			return fmt.Errorf("missing vrouter ASN value")
+		}
+		if _, err := strconv.ParseInt(checkArgs[1], 10, 64); err != nil {
+			return fmt.Errorf("invalid vrouter ASN: %w", err)
+		}
+		checkArgs = checkArgs[2:] // re-slice processed arguments
+	}
+
+	// [peer|neighbor <address>]
+	if args[0] == advertisedKW {
+		if len(checkArgs) == 0 || (checkArgs[0] != peerKW && checkArgs[0] != neighborKW) {
+			return fmt.Errorf("missing `%s` argument", peerKW)
+		}
+		if len(checkArgs) < 2 {
+			return fmt.Errorf("missing peer IP address")
+		}
+		if _, err := netip.ParseAddr(checkArgs[1]); err != nil {
+			return fmt.Errorf("invalid peer IP address: %w", err)
+		}
+	}
+	return nil
+}
+
+func printRouteSummary(out io.Writer, routesPerNode map[string][]*models.BgpRoute) {
+	// sort by node names
+	var nodes []string
+	for node := range routesPerNode {
+		nodes = append(nodes, node)
+	}
+	sort.Strings(nodes)
+
+	// sort routes per node
+	for _, routes := range routesPerNode {
+		// sort routes first by ASN and then by prefix
+		sort.Slice(routes, func(i, j int) bool {
+			return routes[i].RouterAsn < routes[j].RouterAsn || routes[i].Prefix < routes[j].Prefix
+		})
+	}
+
+	w := tabwriter.NewWriter(out, minWidth, 0, padding, paddingChar, 0)
+	fmt.Fprintln(w, "Node\tVRouter\tPrefix\tNextHop\tAge\tAttrs")
+
+	for _, node := range nodes {
+		routes := routesPerNode[node]
+
+		for i, route := range routes {
+			if i == 0 {
+				// print name for first row of node's routes
+				fmt.Fprintf(w, "%s\t", node)
+			} else {
+				// skip name for all rest of the routes
+				fmt.Fprint(w, "\t")
+			}
+
+			r, err := api.ToAgentRoute(route)
+			if err != nil {
+				continue
+			}
+			for _, path := range r.Paths {
+				fmt.Fprintf(w, "%d\t", route.RouterAsn)
+				fmt.Fprintf(w, "%s\t", path.NLRI)
+				fmt.Fprintf(w, "%s\t", nextHopFromPathAttributes(path.PathAttributes))
+				fmt.Fprintf(w, "%s\t", time.Duration(path.AgeNanoseconds).Round(time.Second))
+				fmt.Fprintf(w, "%s\t", path.PathAttributes)
+				fmt.Fprintf(w, "\n")
```

Comment:
Should new line be added if len(path) = 0 and also at line 256 ?

Reply:
that would be kind of an invalid route, but can do that just in case